### PR TITLE
AArch64: Implement iRegLoad/iRegStoreEvaluator(), etc.

### PR DESCRIPTION
**Pull request by knn-k**
This commmit implements iRegLoadEvaluator(), iRegStoreEvaluator(), and
similar evaluators for aarch64.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

File: compiler/aarch64/codegen/TreeEvaluatorTable.hpp [hunk 1]
```diff
@@ -394,21 +394,21 @@
...
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::bRegLoadEvaluator ,	// TR::bRegLoad		// Load byte global register
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::iRegStoreEvaluator ,	// TR::iRegStore		// Store integer global register
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::aRegStoreEvaluator ,	// TR::aRegStore		// Store address global register
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::lRegStoreEvaluator ,	// TR::lRegStore		// Store long integer global register
+    TR::TreeEvaluator::iRegLoadEvaluator, // TR::sRegLoad		// Load short global register
+    TR::TreeEvaluator::iRegLoadEvaluator, // TR::bRegLoad		// Load byte global register
+    TR::TreeEvaluator::iRegStoreEvaluator, // TR::iRegStore		// Store integer global register
+    TR::TreeEvaluator::lRegStoreEvaluator, // TR::aRegStore		// Store address global register
+    TR::TreeEvaluator::lRegStoreEvaluator, // TR::lRegStore		// Store long integer global register
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/TreeEvaluatorTable.hpp]: Why not save the extra level of indirection and have this table reference `commonRegStoreEvaluator` directly instead, and delete the i,a,l, etc. versions?

File: compiler/aarch64/codegen/TreeEvaluatorTable.hpp [hunk 2]
```diff
@@ -394,21 +394,21 @@
...
     TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::callIfEvaluator ,	// TR::callIf		// Call symref if first child evaluates to true.  Other childrem are arguments to the call.
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::iRegLoadEvaluator ,	// TR::iRegLoad		// Load integer global register
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::aRegLoadEvaluator ,	// TR::aRegLoad		// Load address global register
-    TR::TreeEvaluator::unImpOpEvaluator ,        // TODO:ARM64: Enable when Implemented: TR::TreeEvaluator::lRegLoadEvaluator ,	// TR::lRegLoad		// Load long integer global register
+    TR::TreeEvaluator::iRegLoadEvaluator, // TR::iRegLoad		// Load integer global register
+    TR::TreeEvaluator::aRegLoadEvaluator, // TR::aRegLoad		// Load address global register
+    TR::TreeEvaluator::lRegLoadEvaluator, // TR::lRegLoad		// Load long integer global register
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/TreeEvaluatorTable.hpp]: Why not save the extra level of indirection and have this table reference `commonRegLoadEvaluator` instead, and delete the i,a,l, etc. versions?

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 3]
```diff
@@ -364,45 +365,90 @@ OMR::ARM64::TreeEvaluator::loadaddrEvaluator(TR::Node *node, TR::CodeGenerator *
...
+   }
 
 TR::Register *
 OMR::ARM64::TreeEvaluator::GlRegDepsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::GlRegDepsEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   int i;
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: int32_t for consistency with rest of codebase.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

**Comment by knn-k** [comment]
Updated files reflecting the review comments.

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

**Comment by 0xdaryl** [comment]
This is an AArch64-specific change.  A full genie build is not required.

**Pull request merged**